src/app/tools/active_tool.cpp

```diff
@@ -116,8 +119,10 @@ Ink* ActiveToolManager::adjustToolInkDependingOnSelectedInkType(
         id = tools::WellKnownInks::Shading;
         break;
     }
-    if (id)
+    if (id) {
       ink = m_toolbox->getInkById(id);
+      ink->setForcedCopyInk(simpleTurnsCopyColor);
```

Comment:
I made some comments about how we could refactor the implementation of this method...but now I'm not sure that we should actually have this method (along with the wasForcedCopyInk() method). This is because here we are modifying the state of an ink object of the m_toolbox->m_inks map. This map is initialized in the ToolBox constructor, and it seems to me that the idea is that each ink should be immutable...
If we were returning a new instance of an Ink then it shouldn't be a problem...but then I don't know about the  implications of this change.